File: services/payment-service/src/main/java/io/minishop/payment/web/dto/CreatePaymentRequest.java
package io.minishop.payment.web.dto;

import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.PositiveOrZero;

import java.math.BigDecimal;

public record CreatePaymentRequest(
        @NotNull Long orderId,
        @NotNull Long userId,
        @NotNull @PositiveOrZero BigDecimal amount
) {}

File: services/payment-service/src/main/kotlin/io/minishop/payment/web/dto/CreatePaymentRequest.kt
package io.minishop.payment.web.dto

import jakarta.validation.constraints.NotNull
import jakarta.validation.constraints.PositiveOrZero
import java.math.BigDecimal

@JvmRecord
data class CreatePaymentRequest(
    @field:NotNull val orderId: Long?,
    @field:NotNull val userId: Long?,
    @field:NotNull @field:PositiveOrZero val amount: BigDecimal?,
)
